plugins/passwordagentauthplugin/passwordagentplugin.cpp

```diff
@@ -500,9 +526,12 @@ Result PasswordAgentPlugin::beginAuthentication(
     m_polkitResponses.insert(cookie, response);
 
     connect(response, &QDBusPendingCallWatcher::finished, this, [=](QDBusPendingCallWatcher *watcher) {
-        watcher->deleteLater();
-
-        m_polkitResponses.remove(cookie);
+        QHash<QString, PolkitResponse *>::Iterator it = m_polkitResponses.find(cookie);
+        if (it == m_polkitResponses.end()) {
```

Comment:
slight behaviour change here: no watcher->deleteLater() ?

Reply:
Well, in fact it has been displaced just after the find(). I need the find result to quickly return if the authentication has been cancelled already. Then, having the find result at hand, I found more logical to move the deleteLater after the test that the watcher object still exists in m_polkitResponses.